### PR TITLE
Add fields to extract() function

### LINKED ISSUE #765: Document the added fields of extract ()
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/8830

### Which part(s) of the docs might be affected or should be updated? And how?

Update the extract() function here: https://www.risingwave.dev/docs/current/sql-function-datetime/

> support the following fields for extract and date_part:
> millisecond
> microsecond
> epoch
> millennium
> century
> decade
> isoyear
> quarter
> week
> isodow
> julian

### Reference

_No response_

### PR DESCRIPTION
**Pull request by bernscode**
Add fields to extract() function.

## Info
- **Description**: 
Add fields to extract() function.

- **Preview**: 
https://pr-822.d2fbku9n2b6wde.amplifyapp.com/docs/upcoming/sql-function-datetime/

- **Related code PR**: 
https://github.com/risingwavelabs/risingwave/pull/8830

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/765

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

File: docs/sql/functions-operators/sql-function-datetime.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ title: Date and time functions and operators
...
-| extract ( *field* FROM *source* ) → *numeric_output* | Extracts the value of a date or timestamp. <br /> *field* can be year, month, day, hour, minute, second, doy, or dow. <br /> *source* can be the date or timestamp.|`extract(day from date '2022-04-07')` → `7` <br /> `extract(hour from timestamp '2022-04-07 22:00:30')` → `22`|
+| extract ( *field* FROM *source* ) → *numeric_output* | Extracts the value of a date or timestamp. <br /> *field* can be year, month, day, hour, minute, second, doy, dow, millisecond, microsecond, epoch, millennium, century, decade, isoyear, quarter, week, isodow, or julian. <br /> *source* can be the date or timestamp.|`extract(day from date '2022-04-07')` → `7` <br /> `extract(hour from timestamp '2022-04-07 22:00:30')` → `22`|
```
Comment (wangrunji0408) [inline docs/sql/functions-operators/sql-function-datetime.md]: Source can be time and interval as well. some examples:
```
`extract(second from time '22:00:30.123')` → `30.123000`
`extract(day from interval '2 days')` → `2`
```

Comment (bernscode) [reply docs/sql/functions-operators/sql-function-datetime.md]: @wangrunji0408 Thanks for the additional info! I've added this to the latest commit.

**Review by wangrunji0408 — APPROVED** [review]
LGTM, thanks!

**Comment by CharlieSYH** [comment]
@bernscode Please also document the `date_part` function. It's in the same PR as "more fields for extract". Apology for missing out on the function in the doc issue.

**Comment by CharlieSYH** [comment]
> @bernscode Please also document the `date_part` function. It's in the same PR as "more fields for extract". Apology for missing out on the function in the doc issue.

Sorry, already documented. https://github.com/risingwavelabs/risingwave-docs/pull/832

**Pull request merged**